lib/embulk/guess/schema_guess.rb

```diff
@@ -62,11 +62,8 @@ def guess_type(str)
           return "long"
         end
 
-        if str.include?('.')
-          a, b = str.split(".", 2)
-          if a.to_i.to_s == a && b.to_i.to_s == b
-            return "double"
-          end
+        if str =~ /^[-+]?([1-9]\d*|0)(\.\d+)([eE][-+]?[0-9]+)?$/
```

Comment:
> I thought it's okay to "001" can be a number to be honest, but it should be determined by those who know the actual usage more.

Agree.

> To build a regex for that purpose, taking a look at Guava's may help instead of building from scratch. Dropping unnecessary parts from that may be easier.

Thank you, that's much help for me.